collect_app/src/androidTest/java/org/odk/collect/android/feature/external/FormEditActionTest.kt

```diff
@@ -36,61 +32,4 @@ class FormEditActionTest {
         val intent = Intent(Intent.ACTION_EDIT).also { it.data = uri }
         rule.launch(intent, FormEntryPage("One Question"))
     }
-
-    @Test
-    fun whenFormIsNotCurrentProject_showsWarningAndExits() {
```

Comment:
Just wanted to talk through the test restructuring here before reviewing. Is the idea that `FormEditActionTest` checks the "happy path" or simplest case (and most importantly that the intent filter is set up correctly) and that the `FormUriActivityTest` checks the more detailed cases (missing project etc)? It looks like a sort of "feature and component test" breakdown but just wanted to check we were on the same page!

Reply:
`FormEditActionTest` covered the same cases like the new `FormUriActivityTest`. I think `FormEditActionTest` was created as a UI test that requires android emulator because we weren't able to test implicit intents using activity scenario https://github.com/android/android-test/issues/496 that issue will be solved soon.

I kept that one test in `FormEditActionTest` for two reasons:
1. What I mentioned above - currently in `FormUriActivityTest` we only test explicit intents so it's nice to keep one that odes that in an implicit way.
2. It's nice to have one UI test for that scenario (simple scenario) but all the other cases should be tested using activity scenario.

Reply:
Right that all makes sense!